src/pages/DashboardPage.js

```diff
@@ -1,25 +1,65 @@
-import React, {useState} from 'react';
+import React, {useEffect, useState} from 'react';
 import {
   Routes,
   Route
 } from 'react-router-dom';
 import SideBar from '../compontens/SideBar';
 import Title from '../compontens/Title';
-import LatestBlogsPage from './LatestBlogsPage';
-import NewBlogPage from './NewBlogPage';
-import MyBlogsPage from './MyBlogsPage';
-import BlogPage from './BlogPage';
-import { useSelector, useDispatch } from 'react-redux';
-import {addBlog, removeBlog} from '../store';
+import LatestComponent from '../compontens/LatestComponent';
+import NewBlogComponent from '../compontens/NewBlogComponent';
+import MyBlogsComponent from '../compontens/MyBlogsComponent';
+import BlogComponent from '../compontens/BlogComponent';
 import {ImProfile} from 'react-icons/im';
+import { useDispatch, useSelector } from 'react-redux';
+import { addBlog } from '../store';
 
 export default function DashboardPage({setAuth}) {
   const dispatch = useDispatch();
+  const [username, setUsername] = useState("");
 
-  const blogs = useSelector((state)=>{
-    return state.blogs;
+  useEffect(()=>{
+    getProfile();
+    fetchBlogs();
   });
 
+  const fetchBlogs = async ()=>{
+    try {
+      
+      const response = await fetch(`${process.env.REACT_APP_BASE_URL}/blogs/allblogs`, {
+        method: 'GET',
+        headers: {
+          token: localStorage.token
+        }
+      });
+
+      const parsedResponse = await response.json();
+
+      dispatch(addBlog(parsedResponse));
+
+    } catch (error) {
+        console.log(error.message);
+    }
+  }
+
+  const getProfile = async()=>{
+    try {
+  
+      const response = await fetch(`${process.env.REACT_APP_BASE_URL}/user/username`,{
```

Comment:
Again here you see you need to set `method : "GET"`, with axios it would be `.get(url......)` :)